lib/flipper/api/v1/decorators/gate.rb

```diff
@@ -0,0 +1,35 @@
+module Flipper
+  module Api
+    module V1
+      module Decorators
+        class Gate < SimpleDelegator
+          # Public the gate being decorated
+          alias_method :gate, :__getobj__
+
+          # Public: the value for the gate from the adapter.
+          attr_reader :value
+
+          def initialize(gate, value = nil)
+            super gate
+            @value = value
+          end
+
+          def as_json
+            {
+              'key' => gate.key.to_s,
+              'name' => gate.name.to_s,
+              'value' => value_as_json
```

Comment:
I usually drop trailing commas here as ruby doesn't care and then I never have to remember to add a comma to drop in a new key/value. If you think about it when you tweak other stuff go for it. If not, no worries.